src/api/internal.rs

```diff
@@ -322,7 +322,9 @@ impl<T: Pixel> ContextInner<T> {
       let lookahead_frames = self
         .frame_q
         .range(self.next_lookahead_frame - 1..)
-        .filter_map(|(&_input_frameno, frame)| frame.clone())
+        .filter_map(|(&_input_frameno, frame)| {
+          frame.as_ref().map(|f| Arc::clone(f))
```

Comment:
Small nit, you can remove the closure and replace it with `Arc::clone` itself:
```suggestion
          frame.as_ref().map(Arc::clone)
```